### PR TITLE
[ntuple, browser] Use RField visitor to draw histograms

### PR DESCRIPTION
**Pull request by jblomer**
# This Pull request:
Uses the RFieldVisitor infrastructure to draw histograms in the ROOT7 browser.

## Changes or fixes:
- Adds browsing support for all numerical RNTuple types
- Fixes browsing of RNTuple collections

### CODE REVIEW

File: gui/browsable/src/RFieldProvider.hxx [hunk 1]
```diff
@@ -12,43 +12,40 @@
...
 
 using namespace ROOT::Experimental::Browsable;
 
 using namespace std::string_literals;
 
+template<typename T>
+using RField = ROOT::Experimental::RField<T>;
 
-// ==============================================================================================
+namespace {
```
Comment (pcanal) [inline gui/browsable/src/RFieldProvider.hxx:34]: Why use an unnamed in a header file?

File: gui/browsable/src/RFieldProvider.hxx [hunk 2]
```diff
@@ -12,43 +12,40 @@
...
-
-   // virtual ~RFieldProvider() = default;
+class RDrawVisitor : public ROOT::Experimental::Detail::RFieldVisitor {
+private:
+   std::shared_ptr<ROOT::Experimental::Detail::RPageSource> fNtplSource;
+   std::unique_ptr<TH1> fHist;
 
    /** Test collected entries if it looks like integer values and one can use better binning */
-   TH1F *TestHistBuffer(TH1F *hist)
+   void TestHistBuffer()
```
Comment (pcanal) [inline gui/browsable/src/RFieldProvider.hxx:42]: Shouldn't this code be in the 'test' directory?

Comment (jblomer) [reply gui/browsable/src/RFieldProvider.hxx:42]: This is not a unit test but it checks for certain special cases and adjust the histogram accordingly.

File: gui/browsable/src/RFieldProvider.hxx (hunk 1 above)
Comment (jblomer) [reply gui/browsable/src/RFieldProvider.hxx]: Fixed in 58012fe

File: gui/browsable/src/RFieldProvider.hxx [hunk 3]
```diff
@@ -38,132 +42,145 @@ using namespace std::string_literals;
...
+         }
 
-   // virtual ~RFieldProvider() = default;
+         // special case when only integer values in short range - better binning
+         if (is_integer && (max-min < 100)) {
+            max += 2;
+            if (min > 1) min -= 2;
+            int npoints = TMath::Nint(max - min);
+            fHist = std::make_unique<TH1F>(fHist->GetName(), fHist->GetTitle(), npoints, min, max);
+            fHist->SetDirectory(nullptr);
```
Comment (linev) [inline gui/browsable/src/RFieldProvider.hxx:75]: Do I understand correctly that here `fHist` recreated again, providing new `min` and `max` arguments?
If yes - code is wrong. While buffer `fHist->GetBuffer()` will be destroyed together with original histogram.

Comment (jblomer) [reply gui/browsable/src/RFieldProvider.hxx]: Thanks! Should be fixed in [3c02346](https://github.com/root-project/root/pull/8545/commits/3c0234621adba4b020e9d2d110c2c690c3391d38)

**Review by linev — APPROVED** [review]
Now looks ok

**Pull request merged**